org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/lookup/SplitPackageBinding.java

```diff
@@ -258,23 +258,33 @@ public boolean isDeclaredIn(ModuleBinding moduleBinding) {
 
 	@Override
 	public PackageBinding getVisibleFor(ModuleBinding clientModule, boolean preferLocal) {
-		int visibleCount = 0;
-		PlainPackageBinding unique = null;
+		int visibleCountInNamedModules = 0;
+		PlainPackageBinding uniqueInNamedModules = null;
+		PlainPackageBinding bindingInUnnamedModule = null;
 		for (PlainPackageBinding incarnation : this.incarnations) {
 			if (incarnation.hasCompilationUnit(false)) {
 				if (preferLocal && incarnation.enclosingModule == clientModule) {
 					return incarnation;
 				} else {
 					if (clientModule.canAccess(incarnation)) {
-						visibleCount++;
-						unique = incarnation;
+						if (incarnation.enclosingModule.isUnnamed()) { // TODO: check preference
+							bindingInUnnamedModule = incarnation;
+						} else {
+							visibleCountInNamedModules++;
+							uniqueInNamedModules = incarnation;
+						}
 					}
 				}
 			}
 		}
-		if (visibleCount > 1)
+		if (visibleCountInNamedModules > 1)
```

Comment:
As the control structure is growing, please add braces for readability, at least to the outer if-else.